src/lexer/regexFactory.ts

```diff
@@ -165,7 +170,9 @@ export const identifierPattern = ({ first, rest, dashes }: IdentChars = {}): str
   const firstChars = escapeRegExp(first ?? '');
   const restChars = escapeRegExp(rest ?? '');
 
-  const pattern = `[${letter}${firstChars}][${letter}${number}${restChars}]*`;
+  const pattern = allowFirstCharNumber
+    ? `[${firstChars}][${letter}${number}${restChars}]+|[${letter}${number}${restChars}]*[${letter}${restChars}]+[${letter}${number}${restChars}]*`
```

Comment:
So, I see you opted for even more complex solution. When I replace this line with:

```suggestion
    ? `[${letter}${number}${firstChars}][${letter}${number}${restChars}]*`
```

the tests still pass just fine.

Reply:
I was avoiding the all number case but it seems that shouldn't happen with the new number regex